quickfixj-core/src/main/java/quickfix/Session.java

```diff
@@ -2538,6 +2538,16 @@ private void generateLogon(Message otherLogon, int expectedTargetNum) throws Fie
         state.setLogonSent(true);
     }
 
+    private void persist(Message message, int num) throws IOException, FieldNotFound {
+      if (num == 0) {
+          final int msgSeqNum = message.getHeader().getInt(MsgSeqNum.FIELD);
```

Comment:
Could you please move this line inside of the `if (persistMessages)` block since it is not used elsewhere.